src/pocketmine/resourcepacks/ResourcePackManager.php

```diff
@@ -118,6 +124,57 @@ public function __construct(string $path, \Logger $logger){
 		}
 
 		$logger->debug("Successfully loaded " . count($this->resourcePacks) . " resource packs");
+
+		$logger->info("Loading behavior packs...");
+
+		$behaviorStack = $resourcePacksConfig->get("behavior_stack", []);
+		if(!is_array($behaviorStack)){
+			throw new \InvalidArgumentException("\"behavior_stack\" key should contain a list of pack names");
+		}
+
+		foreach($behaviorStack as $pos => $pack){
+			try{
+				$pack = (string) $pack;
+			}catch(\ErrorException $e){
+				$logger->critical("Found invalid entry in behavior pack list at offset $pos of type " . gettype($pack));
+				continue;
+			}
+			try{
+				/** @var string $pack */
+				$packPath = $this->path . DIRECTORY_SEPARATOR . $pack;
+				if(!file_exists($packPath)){
+					throw new ResourcePackException("File or directory not found");
+				}
+				if(is_dir($packPath)){
+					throw new ResourcePackException("Directory behavior packs are unsupported");
+				}
+
+				$newPack = null;
+				//Detect the type of resource pack.
+				$info = new \SplFileInfo($packPath);
+				switch($info->getExtension()){
+					case "zip":
+					case "mcpack":
+						$newPack = new ZippedBehaviorPack($packPath);
+						break;
+				}
+
+				if($newPack instanceof BehaviorPack){
+					$this->behaviorPacks[] = $newPack;
+					$this->uuidList[strtolower($newPack->getPackId())] = $newPack;
+
+					if($newPack->hasClientScripts()){
+						$this->serverHasClientScripts = true;
+					}
+				}else{
+					throw new ResourcePackException("Format not recognized");
+				}
+			}catch(ResourcePackException $e){
+				$logger->critical("Could not load behavior pack \"$pack\": " . $e->getMessage());
+			}
+		}
+
+		$logger->debug("Successfully loaded " . count($this->behaviorPacks) . " behavior packs");
```

Comment:
I don't think it's a good idea to put this here. This will require behaviour packs to be placed inside `resource_packs/` which is going to confuse everyone.

In addition, this entire block of code is repeated.